juju/juju
# [JUJU-1122] Set FORCE-VERSION for uploaded tools;

This PR allows juju to be able to upgrade to a local built snap;

## Checklist


## QA steps

```console
$ sudo snap install juju --channel 2.9/stable --classic
juju (2.9/stable) 2.9.29 from Canonical✓ installed

$ juju version
2.9.29-ubuntu-amd64

$ which juju
/snap/bin/juju

$ juju bootstrap lxd k1
Creating Juju controller "k1" on lxd/localhost
Looking for packaged Juju agent version 2.9.29 for amd64
Located Juju agent version 2.9.29-ubuntu-amd64 at https://streams.canonical.com/juju/tools/agent/2.9.29/juju-2.9.29-linux-amd64.tgz
To configure your system to better support LXD containers, please see: https://github.com/lxc/lxd/blob/master/doc/production-setup.md
Launching controller instance(s) on localhost/localhost...
 - juju-381a10-0 (arch=amd64)
Installing Juju agent on bootstrap instance
Fetching Juju Dashboard 0.8.1
Waiting for address
Attempting to connect to 10.242.98.130:22
Connected to 10.242.98.130
Running machine configuration script...
Bootstrap agent now started
Contacting Juju controller at 10.242.98.130 to verify accessibility...

Bootstrap complete, controller "k1" is now available
Controller machines are in the "controller" model
Initial model "default" added

$ sudo snap remove juju
juju removed

$ sudo snap install juju_2.9.31_amd64.snap --dangerous --classic
juju 2.9.31 installed

$ juju version
2.9.31-ubuntu-amd64

$ which juju
/snap/bin/juju

## => 1.  upgrade to local snap

$ juju upgrade-controller
no prepackaged agent binaries available, using local agent binary 2.9.31.1
best version:
    2.9.31.1
started upgrade to 2.9.31.1

## => 2.  upgrade to local built

$ $GOPATH/bin/juju upgrade-controller --build-agent
no prepackaged agent binaries available, using local agent binary 2.9.31.2 (built from source)
best version:
    2.9.31.2
started upgrade to 2.9.31.2

```

## Documentation changes

No

## Bug reference

https://bugs.launchpad.net/juju/+bug/1812191
